packages/legend/src/Legend.js

```diff
@@ -0,0 +1,147 @@
+import React, { useRef, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { capitalize } from '@vitessce/utils';
+import { select } from 'd3-selection';
+import { scaleLinear } from 'd3-scale';
+import { axisBottom } from 'd3-axis';
+import { getXlinkHref } from './legend-utils';
+
+export const useStyles = makeStyles(() => ({
+  legend: {
+    position: 'absolute',
+    top: '2px',
+    right: '2px',
+    zIndex: '100',
+    fontSize: '10px !important',
+    display: 'flex',
+    flexDirection: 'column',
+    backgroundColor: 'rgba(215, 215, 215, 0.2)',
+    borderRadius: '4px',
+    padding: '2px',
+    lineHeight: '10px !important',
+    '& svg': {
+      top: 0,
+      left: 0,
+      position: 'relative',
+    },
+  },
+}));
+
+
+const titleHeight = 10;
+const rectHeight = 10;
+
+export default function Legend(props) {
+  const {
+    visible: visibleProp,
+    obsType,
+    featureValueType,
+    considerSelections = true,
+    obsColorEncoding,
+    featureSelection,
+    featureValueColormap,
+    featureValueColormapRange,
+    width = 100,
+    height = 40,
```

Comment:
![Screen Shot 2023-03-13 at 3 40 03 PM](https://user-images.githubusercontent.com/43999641/224735304-facace28-0ce7-4bb1-97a8-4f15d5c895b5.png)
Possible to make slightly shorter?

Reply:
Bit of a nit, though not important.  Could make the image shorter?

Reply:
Good point - whether it intersects will depend a bit on the number of characters of the longest label in the index that is used for the heatmap labels, but regardless the element can be more compact to make it less likely to intersect